moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h

```diff
@@ -191,7 +203,8 @@ class JointConstraintSampler : public ConstraintSampler
 
   void clear() override;
 
-  random_numbers::RandomNumberGenerator random_number_generator_; /**< \brief Random number generator used to sample */
+  std::unique_ptr<random_numbers::RandomNumberGenerator>
```

Comment:
Why do you need to change API? Please revert this.

Reply:
Without this, `random_number_generator_` gets default constructed and the assignment operator of `random_numbers::RandomNumberGenerator` is deleted, hence I can't do the following:
```
random_number_generator_ =random_numbers::RandomNumberGenerator(rng_seed);
```
This way it enables us to construct it once. This API change is thus necessary.

Reply:
It would be nicer if you explain why you disagree @v4hn. I'm genuinely curious about how this can be done without API* change. 

*`random_number_generator_` is fully internal, so it is not an API. It will simply have zero effects from the users' perspective.